### PR TITLE
Add FXIOS-7092 [v118] Make FakespotViewController present as resizable sheet

### PR DESCRIPTION
**Pull request by PARAIPAN9**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7092)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15763)

## :bulb: Description

- Implement header to FakespotBottomSheetChildViewController
- Make BottomSheetViewController resizable while animating

https://github.com/mozilla-mobile/firefox-ios/assets/51127880/ee73f537-7fd9-4492-936e-bd26ced83e85

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Fakespot/Components/FakespotBottomSheetChildViewController.swift [hunk 1]
```diff
@@ -0,0 +1,106 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import ComponentLibrary
+import UIKit
+
+class FakespotBottomSheetChildViewController: UIViewController, BottomSheetChild, Themeable {
```
Comment (thatswinnie) [inline Client/Frontend/Fakespot/Components/FakespotBottomSheetChildViewController.swift:9]: Is there a reason to keep this around? It doesn't seem to be used.

Comment (PARAIPAN9) [reply Client/Frontend/Fakespot/Components/FakespotBottomSheetChildViewController.swift:9]: It is very strange that in the Components folder I don't have anything...

**Review by thatswinnie — APPROVED** [review]

File: Client/Frontend/Fakespot/FakespotViewController.swift [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        static let titleLabelFontSize: CGFloat = 17
+        static let headerSpacing = 8.0
+        static let headerBottomMargin = UIEdgeInsets(top: 0, left: 0, bottom: 8, right: 0)
+    }
+
+    var notificationCenter: NotificationProtocol
+    var themeManager: ThemeManager
+    var themeObserver: NSObjectProtocol?
+
+    private lazy var scrollView: UIScrollView = .build()
```
Comment (razvanlitianu) [inline Client/Frontend/Fakespot/FakespotViewController.swift:24]: Pretty cool we can call it with `()`, I've seen in other places like `.build { _ in }` which is not that great.

File: Client/Frontend/Fakespot/FakespotViewController.swift [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import ComponentLibrary
+import UIKit
+import Shared
+
+class FakespotViewController: UIViewController, Themeable {
+    private struct UX {
```
Comment (razvanlitianu) [inline Client/Frontend/Fakespot/FakespotViewController.swift:11]: I'm wondering if this is supposed to be Themeable or ThemeAplicable?

File: Client/Frontend/Fakespot/FakespotViewController.swift [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    private lazy var headerStackView: UIStackView = .build { stackView in
+        stackView.alignment = .center
+        stackView.spacing = UX.headerSpacing
+        stackView.layoutMargins = UX.headerBottomMargin
+        stackView.isLayoutMarginsRelativeArrangement = true
+    }
+
+    private lazy var logoImageView: UIImageView = .build { imageView in
+        imageView.image = UIImage(imageLiteralResourceName: ImageIdentifiers.homeHeaderLogoBall)
```
Comment (razvanlitianu) [inline Client/Frontend/Fakespot/FakespotViewController.swift:38]: Is there a reason the init is not ` UIImage(named:)` ?

**Pull request merged**